cli/flox/doc/doc-build/flox-publish.md

```diff
@@ -0,0 +1,180 @@
+---
+title: FLOX-PUBLISH
+section: 1
+header: "Flox User Manuals"
+...
+
+```{.include}
+./include/experimental-warning.md
+```
+> Feature flag: `publish`
+
+# NAME
+
+flox-publish - Publish local packages for Flox
+
+
+# SYNOPSIS
+
+``` bash
+flox [<general-options>] publish
+     [-d=<path>]
+     [--store-url]
+     [--signing-key]
+     [<package>]...
+```
+
+# DESCRIPTION
+
+Publish the specified `<package>` from the environment in `<path>`,
+and output build artifacts.
+
+## Publishing process
+
+Possible values for `<package>` are all keys under the `build` attribute
+in the `manifest.toml` and you must specify one.
+
+When publishing a package,
+Flox will send the package matadata to the catalog
+and optionally upload the package binaries to the store indicated.
+This allows re-use of the package in other environments.
+
+Flox makes some assertions before publishing, specifically
+
+- The flox environment used to build the package is tracked as a git repo.
+- Tracked files in the repo are all clean.
+- The repo has an upstream origin.
```

Comment:
Is this has an remote named `upstream`? Or has a remote named `origin`? Or just has a remote defined?